src/plugins/dataflow/components/ui/dataflow-program-topbar.tsx

```diff
@@ -50,10 +22,18 @@ interface TopbarProps {
   showRateUI: boolean;
   lastIntervalDuration: number;
   serialDevice: SerialDevice;
+  onRecordDataChange: () => void;
+  programRecordState: number;
+  numNodes: number;
 }
 
 export const DataflowProgramTopbar = (props: TopbarProps) => {
-  const { serialDevice } = props;
+  const { onSerialRefreshDevices, readOnly, serialDevice, programDataRates,
+    dataRate, onRateSelectClick, onRecordDataChange, programRecordState, numNodes } = props;
+
+  const [finished, setFinished] = useState(false);
+  const handleFinished = (isFinished: boolean) => isFinished && setFinished(true);
```

Comment:
This shouldn't be state, it should be determined by `programRecordState` or comparing the number of samples that have been recorded vs the total samples or something similar.